### ISSUE TITLE
Blank NDI inputs

### ISSUE THREAD

**Issue by mossey** [vassbo/freeshow#998]
On windows, Freeshow is able to detect Ndi sources but not able to display the sources. The signal displays the loading animation but is never loaded. 

I have had no success with local Ndi sources running on the same PC as well as the others on other pcs but within the same network. 

The same NDI signals are visible and play on a Mac and on the NDI monitor on the same PC and on the Mac.

I have no issue sending my screens out of Freeshow via NDI. I am able to detect and receive the sources with no hitch

I have checked the windows firewall and made sure to have it allow get inbound traffic but this does not get the video signal to play. 

Additionally, I have reinstalled Ndi tool and Freeshow but the issue persists. 

Important to mention that this was attempted when the laptops are connected via an Ethernet cable. 

**Reproducing**
Running Freeshow version 1.3.0 on windows 11. Go to NDI sources. You may see your Ndi sources but the video stream is not playing at least for my case... 

**Screenshots (Optional)**

**Version (Optional)**
 - OS: [Windows 11]
 - FreeShow Version: [1.3.0]

**Additional context**

**Comment by vassbo** [comment]
It works just fine in my tests on Windows 10, so I'm not sure what the issue could be. 🤔

**Comment by mossey** [comment]
![Screenshot 2024-11-08 152831](https://github.com/user-attachments/assets/507c063e-7e1c-4ce5-b612-f9b6c7286a65)
![Screenshot 2024-11-08 152745](https://github.com/user-attachments/assets/8d78b46a-4e08-445e-858e-94f9f2f2adc3)
What I am experiencing and my firewall tools

**Comment by vassbo** [comment]
👍 Your configuration seems fine. And the inputs are properly detected, I'm not sure why they won't load..

**Comment by mossey** [comment]
Right, FYI, this is the second PC I have tried on with no luck. The PC was running windows 10.

**Comment by AngangGuo** [comment]
I had similar problems with NDI output yesterday.
I found out my NDI version is too old. 
Everything works well after I uninstalled the old version and installed the latest NDI tools.

**Comment by vassbo** [comment]
Thank you @AngangGuo.
@mossey Have you updated to NDI 6?

**Comment by mossey** [comment]
Hi @vassbo I'm on NDI 6![image](https://github.com/user-attachments/assets/bf039e96-79c8-437b-9448-6f205e8ac557)

**Comment by jerm00** [comment]
i have the same issue on linux, using distroav ndi 6.0.0, the streams from obs (testing with this) are detected correctly, but won't display when clicked. freeshow sort of locks up when i attempt it, high cpu usage even after esc/clear all. no other shows will display after attempting to load the ndi stream. i ended up forcing it to close (alt+f2, pkill -9 freeshow) let me know if you need more info.

**Comment by yurividal** [comment]
I have the same issue on Windows 10. I wonder if there is a way to get more verbose logs from Freeshow or the NDI connector.

I'll also try to run Freeshow as Administrator to see if it works

**Comment by yurividal** [comment]
Running as dmin didn't help.

Here is what freeshow logs says:

````
		{
			"time": "2024-11-26T20:26:06.713Z",
			"os": "win32",
			"version": "1.3.2-beta.1",
			"active": {
				"window": "main",
				"page": "show",
				"show": null,
				"edit": {
					"items": []
				}
			},
			"drawer": {
				"active": "media"
			},
			"type": "unhandledrejection",
			"source": "See stack",
			"message": "Could not start video source"
		}
````

Perhaps you can make that stack trace print the exception in the logs, which will probably give us a bit more info.

Just out of curiosity, is your Win10 PC Intel or AMD?

**Comment by vassbo** [comment]
AMD, I think the video source can't start because the input feed is invalid. The problem is most likely the NDI package!
https://github.com/vassbo/grandiose

**Comment by vassbo** [comment]
I have tested on both Windows 10 & 11, and it works for me (Both AMD).
The inputs load fine, and the one webcam I set to a test pattern loads (the other NDI webcams gives this result as I have not set any input: https://github.com/ChurchApps/FreeShow/issues/1061#issue-2695980019)

**Comment by vreykin** [comment]
I'm also new here, and thank you so much for this fix, hope this gets fixed soon.

**Comment by vassbo** [comment]
@Vitlyr Do you have the same issue in Media>Screens>NDI? Are you on Windows?

**Comment by claudedeslandes** [comment]
I am using a Windows 11 machine.

**Comment by vreykin** [comment]
@vassbo Yep, I am using windows 11. Unfortunately the private network solution didn't really work for me.

**Comment by vassbo** [comment]
There has been some changes to NDI inputs in the latest beta, would someone like to test it out?: https://github.com/ChurchApps/FreeShow/releases/tag/v1.4.9-beta.1

**Comment by lost101** [comment]
@vassbo  I have the same issue on 3 different W11 PC. I have been using NDI for about 4 years but on Freeshow  is a no go. One of the  PC is Intel I9/nvidia and second one is AMD integrated gpu and third  I5 with integrated graphics. 
Network is set to private, firewall allows traffic for Freeshow, sources  work on NDI Monitor on all devices.

I have installed the latest beta  and  happy to report that it does not crash the app.  However no video when activated and thumbnail  loads  sometimes when NDI source selected.

**Comment by jzongker** [comment]
@lost101 - 1.4.9-beta.2 was just published.  Would you mind checking the NDI again?  I was finally able to replicate the issue with one of my NDI sources.  It turned out to be non-video frames are occasionally received (audio, metadata, network issues?), which was treated as an error and would terminate the connection.  Adding handling for these fixed it for me.  I'm hoping it will for you too.

https://github.com/ChurchApps/FreeShow/commit/11488e76574a4957c4e8118b71e25e2a7478afed

**Comment by lost101** [comment]
@jzongker  Installed 1.4.9 beta 2. Reset all settings and found some improvements, but very glitchy.  All tests performed on W11 AMD, with hardware acceleration enabled/disabled.

I have noticed that most of the time preview window does not have video but  video is present in the output. NDI seems to be dropping frames.  When it does work, there is delay in the video compared to NDI monitor  of about 300-400 milliseconds.

I have tested with  3 different NDI sources. One is a Test Pattern generator , one is NDI VLC player and one of them is a Screen Capture HX. (all by Newtek) 
 Playback of full NDI is working with dropped frames, and at times it freezes Freeshow. 
It seems that Freeshow can not decode NDI HX at all.  I h  tested with iPhone NDI Cam, NDI bridge and all did not generate  any image. 
Thank you for your work and improvements.

<img width="3640" height="1920" alt="Image" src="https://github.com/user-attachments/assets/b1edbe1e-41a2-4c0f-9aed-e5eb559dad3e" />
<img width="2560" height="1392" alt="Image" src="https://github.com/user-attachments/assets/0afbfe6c-aa7f-4a52-a718-2f9fd29c02a1" />





Additional Testing: On Fedora, same hardware  same problems or worse. NDI screen capture does not work at all.

**Comment by vassbo** [comment]
@lost101 1.4.9 is released with updates to NDI if you would like to check it out.

**Comment by lost101** [comment]
@vassbo Nice progress with this new release!
Tested on W11, FreeShow 1.4.9, NDI 6.2.1.0 

NDI HX2 working now, but FreeShow is grabbing Low Bandwidth stream. That applies to full NDI, HX2, Bridge, VLC and NDI HX2 iPhone camera. 
There is  inconsistent latency and many times I have to toggle output to regain video. The output preview window sometimes will play the NDI stream but no video is being sent to the screen and vice versa. 

I appreciate your work, God bless!

**Comment by yurividal** [comment]
FYI, if you are running NDI on windows 10 or 11, you might be seeing NDI latency and poor video quality due to a bug recently introduced by microsoft.

https://github.com/DistroAV/DistroAV/issues/1337

I was seeing this issue on all my NDI feeds across my local network. The temporary workarround is to change the NDI settings to use TCP only, instead of UDP

**Comment by paulpv** [comment]
The official NDI writeup on the current Windows 10 and 11 issue is at:
https://docs.ndi.video/all/faq/common-issues/traffic-drops-after-windows-update

**Unofficially**, a Windows Update patch for this is estimated to be released 9/9.

**Comment by vassbo** [comment]
Another report from the Facebook group:
<img width="602" height="592" alt="Image" src="https://github.com/user-attachments/assets/e975f93b-90da-4ed7-97e8-e30fed697f61" />
